src/cpp/session/SessionSourceDatabaseSupervisor.cpp

```diff
@@ -354,18 +351,29 @@ bool reclaimOrphanedSession(FilePath* pSessionDir)
 
    BOOST_FOREACH(const FilePath& sessionDir, sessionDirs)
    {
+      // if the suspend file exists, this session is only sleeping, not dead
+      if (sessionSuspendFilePath(sessionDir).exists())
```

Comment:
It adds to the complexity of the change, but yes, with some plumbing it's possible to distinguish a suspend for restart from an ordinary timeout-based (or server-initiated) suspend. Done here: https://github.com/rstudio/rstudio/commit/c5bd4974bcaac6d65dca05a455580800b36f9d0b